src/Microsoft.AspNet.Authentication.OAuth/OAuthAuthenticationHandler.cs

```diff
@@ -156,17 +179,30 @@ protected override AuthenticationTicket AuthenticateCore()
             string oauthTokenResponse = await response.Content.ReadAsStringAsync();
 
             JObject oauth2Token = JObject.Parse(oauthTokenResponse);
-            return new TokenResponse(oauth2Token);
+            return new OAuthTokenResponse(oauth2Token);
         }
 
-        protected virtual async Task<AuthenticationTicket> GetUserInformationAsync(AuthenticationProperties properties, TokenResponse tokens)
+        protected virtual async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, OAuthTokenResponse tokens)
         {
-            var context = new OAuthGetUserInformationContext(Context, Options, Backchannel, tokens)
+            if (typeof(TAuthenticationContext) != typeof(OAuthAuthenticatedContext))
+            {
+                throw new InvalidOperationException();
```

Comment:
We need a nice message here (well, in the .resx file actually). Any idea? :smile: